addons/mail/models/mail_activity.py

```diff
@@ -55,27 +55,25 @@ def default_get(self, fields):
         ('warning', 'Alert'),
         ('danger', 'Error')], string="Decoration Type",
         help="Change the background color of the related activities of this type.")
-    res_model_id = fields.Many2one(
-        'ir.model', 'Model', index=True,
-        domain=['&', ('is_mail_thread', '=', True), ('transient', '=', False)],
+    res_model = fields.Selection(selection=_get_model_selection, string="Model", index=True,
```

Comment:
As we don't have a m2o to a model anymore, shouldn't we override unlink of ir model to remove linked activity types ?

Reply:
Ok done sir

Reply:
Remove the activity type or set null ?

Reply:
Would keep the same behavior, aka removing it. We have global and model-based activities, would be strange that a specific one become global.